mypy/build.py

```diff
@@ -710,7 +715,7 @@ def read_with_python_encoding(path: str, pyversion: Tuple[int, int]) -> str:
             source_bytearray.decode(encoding)
         except LookupError as lookuperr:
             raise DecodeError(str(lookuperr))
-        return source_bytearray.decode(encoding)
+        return source_bytearray.decode(encoding), hashlib.md5(source_bytearray).hexdigest()
```

Comment:
This is incorrect if a BOM marker is present, since on line 705 above the BOM marker is removed from source_bytearray.